### PR TITLE
Update vendored azure-sdk-for-go

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Updating to a recent version of Azure Storage SDK to be
able to **patch some memory leaks** through configurable HTTP client
change added to the registry recently.

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
 	"Deps": [
 		{
 			"ImportPath": "github.com/Azure/azure-sdk-for-go/storage",
-			"Comment": "v1.2-334-g95361a2",
-			"Rev": "95361a2573b1fa92a00c5fc2707a80308483c6f9"
+			"Comment": "v6.0.0-beta",
```
Comment (RichardScothern) [inline Godeps/Godeps.json]: I don't like using a package marked as beta into a production system.

Is there something between v1.2 and v6 that is more stable?

Comment (ahmetalpbalkan) [inline Godeps/Godeps.json]: @RichardScothern It's all stable, don't mind the name. 😸 I'm not sure why we still keep adding `-beta`.

**Comment by RichardScothern** [comment]
LGTM

**Comment by ahmetalpbalkan** [comment]
More love for this PR?

**Comment by ahmetalpbalkan** [comment]
Actually, let's not merge it yet, I discovered some bugs in our SDK that I need to first patch in the azure-sdk-for-go repo.

**Comment by ahmetalpbalkan** [comment]
@RichardScothern ready to go now!

 I updated the vendored bits to include [this commit](https://github.com/Azure/azure-sdk-for-go/pull/444/commits/6935fa3ae398a0fa90cc3f83da583e3691c86443) which was giving a nil dereference error on some improper `resp.Body.Close()` which was causing some registry requests to panic.

**Comment by dmcgowan** [comment]
LGTM, update name now?

**Comment by ahmetalpbalkan** [comment]
@dmcgowan updated.

**Comment by ahmetalpbalkan** [comment]
2 LGTMs, so merge maybe? 😸

**Pull request merged**